File: ejercicio2/file_reader.rb
class FileReader
  def self.read_cases(file_path)
    cases = []
    current_case = nil
    File
      .open(file_path)
      .each do |line|
        if line.start_with?("#")
          current_case = line[/\d+/].to_i
        elsif !line.strip.empty?
          cases << [current_case, line.strip.to_i]
        end
      end
    cases
  end
end

File: ejercicio3/file_reader.py
class FileReader:
  def read_cases(file_path):
      cases = []
      current_case = None
      with open(file_path) as f:
          for line in f:
              if line.startswith('#'):
                  current_case = line.split()[-1]
              elif line.strip():
                  cases.append([current_case, line.strip()])
      return cases
